pkg/agent/execbridge.go

```diff
@@ -52,20 +67,26 @@ func (iface *execPluginInterface) Request(
 		iface.runner.recordResourceChange(*lastPermit, resp.Permit, iface.runner.global.metrics.schedulerApprovedChange)
 	}
 
-	successful := func() bool {
+	successful, responseType := func() (bool, scalingResponseType) {
 		if err != nil { // request is failed
-			return false
+			return false, scalingResponseTypeFailed
 		}
 		if resp.Permit == target { // request is fully approved by the scheduler
-			return true
+			return true, scalingResponseTypeApproved
 		}
 		if lastPermit != nil && *lastPermit != resp.Permit { // request is partially approved by the scheduler
-			return true
+			return true, scalingResponseTypePartiallyApproved
 		}
-		return false // scheduler denied the request
+		return false, scalingResponseTypeDenied // scheduler denied the request
 	}()
 	iface.runner.status.update(iface.runner.global, func(ps podStatus) podStatus {
 		if !successful {
+			switch responseType {
```

Comment:
I don't think this needs to be inside the callback.

Reply:
moved away from the callback